snap/snapcraft.yaml

```
@@ -86,10 +102,22 @@ parts:
       - dotnet-hostfxr-8.0
     override-stage: |
       craftctl default
+
+      if [ "${CRAFT_ARCH_BUILD_FOR}" = "amd64" ]; then
+        INTERPRETER="/snap/dotnet/current/lib64/ld-linux-x86-64.so.2"
+        RPATH="/snap/dotnet/current/usr/lib/x86_64-linux-gnu"
+      elif [ "${CRAFT_ARCH_BUILD_FOR}" = "arm64" ]; then
+        INTERPRETER="/snap/dotnet/current/lib/ld-linux-aarch64.so.1"
+        RPATH="/snap/dotnet/current/usr/lib/aarch64-linux-gnu:/snap/dotnet/current/lib/aarch64-linux-gnu"
+      else
+        echo "Unknown architecture (${CRAFT_ARCH_BUILD_FOR})"
```

nitpick: The architecture is probably known, but unsupported.

```suggestion
        echo "Unsupported architecture (${CRAFT_ARCH_BUILD_FOR})"
```